packages/ui/src/common/providers/modal/provider.tsx

```diff
@@ -24,8 +25,15 @@ export const ModalContextProvider = (props: Props) => {
         setModalData(modalCall.data)
       }
     },
-    hideModal: () => {
-      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames)) {
+    hideModal: (showConfirmOrAny = false) => {
+      const preferShowConfirm = typeof showConfirmOrAny === 'boolean' ? showConfirmOrAny : false
+      const [state] = currentModalMachine ?? []
+      if (
+        !preferShowConfirm ||
```

Comment:
Similarly to https://github.com/Joystream/pioneer/pull/3938#discussion_r1105509469

Please remove `showConfirmOrAny`. Especially since typescript won't allow passing this parameter.

```suggestion
    hideModal: () => {
      const [state] = currentModalMachine ?? []
      if (
```